commit a19fb5ec351ca3a52501b3a3426deba5806ebb83
Author: Vincent Bernat <bernat@luffy.cx>
Date:   Sun Feb 20 12:17:57 2011 +0100

    vrrp: on shutdown, release sockets later to be able to send shutdown packet
    
    Each VRRP instance should be able to send a shutdown packet when the
    instance is terminated. However, the socket pool was released early
    and the socket was closed too soon. We release the socket pool later
    to avoid this problem.

diff --git a/keepalived/vrrp/vrrp_daemon.c b/keepalived/vrrp/vrrp_daemon.c
index b44398e..555b55f 100644
--- a/keepalived/vrrp/vrrp_daemon.c
+++ b/keepalived/vrrp/vrrp_daemon.c
@@ -52,7 +52,6 @@ stop_vrrp(void)
 {
 	/* Destroy master thread */
 	signal_handler_destroy();
-	free_vrrp_sockpool(vrrp_data);
 	thread_destroy_master(master);
 
 	/* Clear static entries */
@@ -70,6 +69,7 @@ stop_vrrp(void)
 
 	/* Clean data */
 	free_global_data(data);
+	free_vrrp_sockpool(vrrp_data);
 	free_vrrp_data(vrrp_data);
 	free_vrrp_buffer();
 
